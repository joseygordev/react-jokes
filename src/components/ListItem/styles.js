import theme from '../../config/theme';

const lineItem = {
  background: theme.colors.white,
  borderRadius: '0.5rem',
  boxShadow: '0.125rem 0.125rem 0.25rem rgba(0, 0, 0, 0.25)',
  display: 'flex',
  justifyContent: 'space-between',
  padding: theme.spacing.padding * 2,
  margin: `${theme.spacing.margin}px auto`,
  width: '100%',
  flexDirection: 'column',
  borderLeft: `1rem solid ${theme.colors.secondary}`
};

export default {
  listLine: {
    ...lineItem
  },
  listButton: {
    ...lineItem,
  },
  listButtonHover: {
    opacity: .5,
    cursor: 'pointer',
  },
  listTitle: {
    fontWeight: 'bold',
    fontSize: '1.5rem',
    margin: 0,
    textTransform: 'capitalize',
  },
  listContent: {
    fontSize: '1.25rem',
    textTransform: 'capitalize',
    margin: `${theme.spacing.margin}px 0px 0px`,
    textAlign: 'left',
  },
};