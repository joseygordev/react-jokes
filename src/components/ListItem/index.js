import React, {useState} from 'react';
import PropTypes from 'prop-types';

// Styles
import styles from './styles';

const ListItem = ({item, onClick}) => {
  const [hoverActive, setHoverActive] = useState(false);

  const renderContent = () => {
    return (
      <React.Fragment>
        {item.title && <h2 style={styles.listTitle}>{item.title}</h2>}
        {item.content && <p style={styles.listContent}>{item.content}</p>}
      </React.Fragment>
    );
  };

  const render = () => {
    if (onClick) {
      return (
        <button 
          style={{...styles.listButton, ...(hoverActive && styles.listButtonHover)}}
          onMouseEnter={() => setHoverActive(true)}
          onMouseLeave={() => setHoverActive(false)} 
          data-testid="button"
          type="button"
          onClick={() => onClick(item)}>
          {renderContent()}
        </button>
      );
    }
     
    return (
      <div style={styles.listLine}>
        {renderContent()}
      </div>
    );
    
  };

  return render();
};

ListItem.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string,
    content: PropTypes.string,
  }),
  onClick: PropTypes.func,
};
ListItem.defaultProps = {
  item: {
    title: '',
    content: '',
  }
};

export default ListItem;
