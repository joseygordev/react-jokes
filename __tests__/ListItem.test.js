import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/dom';
import { render } from "@testing-library/react";
import ListItem from '../src/components/ListItem';

describe('ListItem', () => {
  const item = {
    title: 'My title',
    content: 'My content'
  };

  it('Should return the ListItem component without any param', () => {
    const tree = renderer
      .create(
        <ListItem />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should return the ListItem component without onClick param', () => {
    const tree = renderer
      .create(
        <ListItem items={item} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should return the ListItem component with onClick param', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<ListItem items={item} onClick={onClick} />);
    
    const button = getByTestId('button');
    fireEvent.click(button);
  });
});
