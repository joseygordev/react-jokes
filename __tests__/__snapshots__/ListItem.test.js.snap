// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Should return the ListItem component without any param 1`] = `
<div
  style={
    Object {
      "background": "#FFFFFF",
      "borderLeft": "1rem solid #8a8a8a",
      "borderRadius": "0.5rem",
      "boxShadow": "0.125rem 0.125rem 0.25rem rgba(0, 0, 0, 0.25)",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "margin": "8px auto",
      "padding": 16,
      "width": "100%",
    }
  }
>
  
  
</div>
`;

exports[`ListItem Should return the ListItem component without onClick param 1`] = `
<div
  style={
    Object {
      "background": "#FFFFFF",
      "borderLeft": "1rem solid #8a8a8a",
      "borderRadius": "0.5rem",
      "boxShadow": "0.125rem 0.125rem 0.25rem rgba(0, 0, 0, 0.25)",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "margin": "8px auto",
      "padding": 16,
      "width": "100%",
    }
  }
>
  
  
</div>
`;
